File: wrappers/dotnet/indy-sdk-dotnet-test/Util/EnvironmentUtils.cs
using System;
using System.IO;
using System.Runtime.InteropServices;

namespace Hyperledger.Indy.Test
{
    static class EnvironmentUtils
    {
        public static string GetTestPoolIP()
        {
            var testPoolIp = Environment.GetEnvironmentVariable("TEST_POOL_IP");
            return testPoolIp != null ? testPoolIp : "127.0.0.1";
        }

        public static string GetUserHomePath()
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                return Environment.GetEnvironmentVariable("HOME");
            else
                return Environment.GetEnvironmentVariable("USERPROFILE");
        }

        public static string GetIndyHomePath()
        {
            return Path.Combine(GetUserHomePath(), ".indy_client");
        }

        public static string GetIndyHomePath(string filename)
        {
            return Path.Combine(GetIndyHomePath(), filename);
        }

        public static string GetTmpPath()
        {
            return Path.Combine(Path.GetTempPath(), "indy_client");
        }

        public static string GetTmpPath(string filename)
        {
            return Path.Combine(GetTmpPath(), filename);
        }

    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/utils/EnvironmentUtils.java
package org.hyperledger.indy.sdk.utils;

import org.apache.commons.io.FileUtils;

public class EnvironmentUtils {
    static String getTestPoolIP() {
        String testPoolIp = System.getenv("TEST_POOL_IP");
        return testPoolIp != null ? testPoolIp : "127.0.0.1";
    }

    public static String getIndyHomePath() {
        return FileUtils.getUserDirectoryPath() + "/.indy_client/";
    }

    public static String getIndyHomePath(String filename) {
        return getIndyHomePath() + filename;
    }

    public static String getTmpPath() {
        return FileUtils.getTempDirectoryPath() + "/indy_client/";
    }

    public static String getTmpPath(String filename) {
        return getTmpPath() + filename;
    }
}
